File: Backend/StudentEnrollment/StudentEnrollment/Models/Student.cs
using System.ComponentModel.DataAnnotations;

namespace StudentEnrollment.Models
{
    public class Student
    {
        public int StudentId { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        public DateTime DateOfBirth { get; set; }

        [Required]
        public string Gender { get; set; }

        [Required]
        public string Nationality { get; set; }

        // Contact information
        [Required]
        [Phone] // Validate phone number format
        public string MobilePhone { get; set; } // Mobile number (personal)

        [Required]
        [EmailAddress] // Validate email format
        public string Email { get; set; }

        [Required]
        public string Address { get; set; }

        // Enrollment Information
        public DateTime EnrollmentDate { get; set; }=  DateTime.Now;// Enrollment date should also be required

        [Required]
        public string Course { get; set; }

        // New field to track the student's year
        public int Year { get; set; } = 1;  // Default value is 1 when enrolling


        // Academic Information
        public double GPA { get; set; }

        // Parent/Guardian information
        [Required]
        public string FatherName { get; set; }

        [Required]
        public string MotherName { get; set; }

        [Required]
        [Phone] // Validate phone number format
        public string FatherPhone { get; set; }

        [Required]
        [Phone] // Validate phone number format
        public string MotherPhone { get; set; }

        // Profile picture (optional)
        public string ProfilePictureUrl { get; set; }
    }
}

File: Frontend/student-enrollment/src/app/Model/student.ts
// src/app/models/student.ts

export class Student {
  studentId: number;
  firstName: string;
  lastName: string;
  dateOfBirth: string|null; // This can be a string that represents the date in ISO format
  gender: string;
  nationality: string;
  mobilePhone: string;
  email: string;
  address: string;
  enrollmentDate: string|null; // This can be a string that represents the date in ISO format
  course: string;
  year: number;
  gpa: number;
  fatherName: string;
  motherName: string;
  fatherPhone: string;
  motherPhone: string;
  profilePictureUrl?: string; // Optional field

  constructor(
    studentId: number,
    firstName: string,
    lastName: string,
    dateOfBirth: string,
    gender: string,
    nationality: string,
    mobilePhone: string,
    email: string,
    address: string,
    enrollmentDate: string,
    course: string,
    year: number,
    gpa: number,
    fatherName: string,
    motherName: string,
    fatherPhone: string,
    motherPhone: string,
    profilePictureUrl?: string // Optional field
  ) {
    this.studentId = studentId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.dateOfBirth = dateOfBirth;
    this.gender = gender;
    this.nationality = nationality;
    this.mobilePhone = mobilePhone;
    this.email = email;
    this.address = address;
    this.enrollmentDate = enrollmentDate;
    this.course = course;
    this.year = year;
    this.gpa = gpa;
    this.fatherName = fatherName;
    this.motherName = motherName;
    this.fatherPhone = fatherPhone;
    this.motherPhone = motherPhone;
    this.profilePictureUrl = profilePictureUrl;
  }
}
